### ISSUE TITLE
Ctrl-Click Soldiers to form group selections causes a CTD (FMT Format Error)

### ISSUE THREAD

**Issue by BeornTB** [OpenApoc/OpenApoc#932]
[CTD ctrl click.zip](https://github.com/OpenApoc/OpenApoc/files/5023559/CTD.ctrl.click.zip)

I tried to select soldiers on screen by ctrl-clicking them individually. CTD each time.

**Comment by Atrosha** [comment]
fmt format error

backtrace

```
terminate called after throwing an instance of 'fmt::v7::format_error'
  what():  invalid type specifier
--Type <RET> for more, q to quit, c to continue without paging--c

Thread 1 "OpenApoc" received signal SIGABRT, Aborted.
0x00007ffff772b9e5 in raise () from /lib64/libc.so.6
(gdb) bt
#0  0x00007ffff772b9e5 in raise () from /lib64/libc.so.6
#1  0x00007ffff7714895 in abort () from /lib64/libc.so.6
#2  0x00007ffff7ab8961 in __gnu_cxx::__verbose_terminate_handler ()
    at ../../../../libstdc++-v3/libsupc++/vterminate.cc:95
#3  0x00007ffff7ac444c in __cxxabiv1::__terminate (handler=<optimized out>)
    at ../../../../libstdc++-v3/libsupc++/eh_terminate.cc:48
#4  0x00007ffff7ac44b7 in std::terminate ()
    at ../../../../libstdc++-v3/libsupc++/eh_terminate.cc:58
#5  0x00007ffff7ac4769 in __cxxabiv1::__cxa_throw (obj=obj@entry=0x1afaca0, 
    tinfo=0xb0c828 <typeinfo for fmt::v7::format_error>, dest=
    0xa2f390 <fmt::v7::format_error::~format_error()>)
    at ../../../../libstdc++-v3/libsupc++/eh_throw.cc:95
#6  0x0000000000478b66 in fmt::v7::detail::int_writer<std::back_insert_iterator<fmt::v7::detail::buffer<char> >, char, unsigned int>::on_error (
    this=<optimized out>)
    at /home/atrosha/OpenApoc/dependencies/fmt/include/fmt/format.h:712
#7  fmt::v7::detail::handle_int_type_spec<fmt::v7::detail::int_writer<std::back_insert_iterator<fmt::v7::detail::buffer<char> >, char, unsigned int>&> (
    spec=<optimized out>, handler=...)
    at /home/atrosha/OpenApoc/dependencies/fmt/include/fmt/format.h:1248
#8  0x000000000047c8bb in fmt::v7::detail::arg_formatter_base<std::back_insert_iterator<fmt::v7::detail::buffer<char> >, char, fmt::v7::detail::error_handler>::write_int<unsigned int> (spec=..., value=<optimized out>, this=<optimized out>)
--Type <RET> for more, q to quit, c to continue without paging--c
    at /home/atrosha/OpenApoc/dependencies/fmt/include/fmt/format.h:1946
#9  fmt::v7::detail::arg_formatter_base<std::back_insert_iterator<fmt::v7::detail::buffer<char> >, char, fmt::v7::detail::error_handler>::operator()<unsigned int, 0> (value=<optimized out>, this=<optimized out>) at /home/atrosha/OpenApoc/dependencies/fmt/include/fmt/format.h:1948
#10 fmt::v7::printf_arg_formatter<std::back_insert_iterator<fmt::v7::detail::buffer<char> >, char>::operator()<unsigned int, 0> (value=<optimized out>, this=<optimized out>) at /home/atrosha/OpenApoc/dependencies/fmt/include/fmt/printf.h:265
#11 fmt::v7::visit_format_arg<fmt::v7::printf_arg_formatter<std::back_insert_iterator<fmt::v7::detail::buffer<char> >, char>, fmt::v7::basic_printf_context<std::back_insert_iterator<fmt::v7::detail::buffer<char> >, char> > (arg=..., vis=...) at /home/atrosha/OpenApoc/dependencies/fmt/include/fmt/core.h:1164
#12 fmt::v7::basic_printf_context<std::back_insert_iterator<fmt::v7::detail::buffer<char> >, char>::format<fmt::v7::printf_arg_formatter<std::back_insert_iterator<fmt::v7::detail::buffer<char> >, char> > (this=0x7fffffffcbe0) at /home/atrosha/OpenApoc/dependencies/fmt/include/fmt/printf.h:594
#13 0x0000000000553010 in fmt::v7::detail::vprintf<char, fmt::v7::basic_printf_context<std::back_insert_iterator<fmt::v7::detail::buffer<char> >, char> > (args=..., format=..., buf=warning: RTTI symbol not found for class 'fmt::v7::basic_memory_buffer<char, 500ul, std::allocator<char> >'
...) at /home/atrosha/OpenApoc/dependencies/fmt/include/fmt/core.h:547
#14 fmt::v7::vsprintf<fmt::v7::basic_string_view<char>, char> (args=..., format=..., format=...) at /home/atrosha/OpenApoc/dependencies/fmt/include/fmt/printf.h:638
#15 0x000000000056b33c in fmt::v7::sprintf<std::basic_string_view<char, std::char_traits<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, char const*, int, int, int, char> (format=Python Exception <class 'gdb.error'> value has been optimized out: 
<synthetic pointer>, format=Python Exception <class 'gdb.error'> value has been optimized out: 
<synthetic pointer>) at /usr/include/c++/10/string_view:242
#16 OpenApoc::format<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, int, char const*, int&, int&, int&> (fmt=...) at /home/atrosha/OpenApoc/library/strings_format.h:11
#17 OpenApoc::BattleView::handleMouseDown (this=<optimized out>, e=<optimized out>) at /home/atrosha/OpenApoc/game/ui/tileview/battleview.cpp:3656
#18 0x000000000057aeb2 in OpenApoc::BattleView::eventOccurred (this=0x2ae2620, e=0x2ae5f90) at /home/atrosha/OpenApoc/game/ui/tileview/battleview.cpp:2880
#19 0x0000000000495901 in OpenApoc::Framework::processEvents (this=0xca2270) at /usr/include/c++/10/bits/unique_ptr.h:421
#20 0x0000000000496525 in OpenApoc::Framework::run (this=this@entry=0xca2270, initialStage=std::shared_ptr<OpenApoc::Stage> (use count 1, weak count 1) = {...}) at /home/atrosha/OpenApoc/framework/framework.cpp:452
#21 0x0000000000469f03 in main (argc=-9296, argv=0x7fffffffdee8) at /usr/include/c++/10/bits/shared_ptr_base.h:759
```

**Comment by Andy51** [comment]
I have checked this was fixed in 7a31e4a52 (released as 0.20220126)

**Comment by FilmBoy84** [comment]
> I have checked this was fixed in [7a31e4a](https://github.com/OpenApoc/OpenApoc/commit/7a31e4a527f0c92f21bfa61764f6df3b945dd371) (released as 0.20220126)

Many thanks for confirming fixed
Closing 🍻
